api/test/jest-config.json

```diff
@@ -31,5 +31,10 @@
   "coveragePathIgnorePatterns": [
     "<rootDir>/node_modules/",
     "<rootDir>/test/"
-  ]
+  ],
+  "globals": {
+    "ts-jest": {
+      "isolatedModules": true
```

Comment:
what does this do, and why do we need it?

Reply:
It is meant to help with memory usage by compiling code on a module basis instead of this is sparta style.
I found it helpful on my machine so letting this here.

Let me know if you want to remove it anyway

Reply:
it seems to have some drawbacks: https://huafu.github.io/ts-jest/user/config/isolatedModules
have you looked into those?

Reply:
Yes I did, project tsc keeps working on the IDE so we still have type safety writing tests. I fail recognising more caveats here so let me know if something smells bad so I can change it back